src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -147,17 +153,17 @@ protected AcquireTokenHandlerBase(RequestData requestData)
 
                 if (ResultEx == null || ResultEx.Exception != null)
                 {
-                    if (PlatformPlugin.BrokerHelper.CanInvokeBroker)
+                    if (brokerHelper.CanInvokeBroker)
                     {
-                        ResultEx = await PlatformPlugin.BrokerHelper.AcquireTokenUsingBroker(brokerParameters).ConfigureAwait(false);
+                        ResultEx = await brokerHelper.AcquireTokenUsingBroker(brokerParameters).ConfigureAwait(false);
                     }
                     else
                     {
                         await this.PreTokenRequest().ConfigureAwait(false);
                         // check if broker app installation is required for authentication.
                         if (this.BrokerInvocationRequired())
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 control flow statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS134)